directives/point_of_interest.py

```diff
@@ -32,50 +40,114 @@ class PointOfInterest(Directive):
     has_content = True
     final_argument_whitespace = True
     option_spec = {
+        'id': directives.unchanged,
         'title': directives.unchanged,
         'previous': directives.unchanged,
         'next': directives.unchanged,
         'hidden': directives.flag,
         'class' : directives.class_option,
+        'height': directives.length_or_percentage_or_unitless,
+        'columns': directives.unchanged,
+        'bgimg':directives.uri,
     }
 
     def run(self):
         self.assert_has_content()
+        env = self.state.document.settings.env
+        if not hasattr(env, 'poi_all'):
+            env.poi_all = {}
         if len(self.arguments) == 0:
             return [nodes.container()]
-        name = self.arguments[0]
-        title_text = ''
-        if 'title' in  self.options:
+
+        if 'title' in self.options:
+            if 'id' in self.options:
+                raise SphinxError(u'Point of interest options can\'t contain both "title" and "id"; one of them should be provided as an argument instead')
+            name = self.arguments[0]
             title_text = self.options['title']
+        else:
+            name = self.options.get('id')
+            if not name:
+              choices = string.ascii_lowercase + string.digits
```

Comment:
Indentation of only 2 spaces when it should be 4?

Reply:
Hups, fixed!